SoftLayer/CLI/virt/create_options.py

```diff
@@ -18,56 +18,121 @@ def cli(env):
+    return flavor_table
+
+
+def _get_cpu_table(create_options):
+    cpu_table = formatting.Table(['cpu', 'value'], title='CPUs')
+    cpu_table.sortby = 'cpu'
+    cpu_table.align = 'l'
+    standard_cpus = [int(x['template']['startCpus']) for x in create_options['processors']
                      if not x['template'].get('dedicatedAccountHostOnlyFlag',
                                               False)
                      and not x['template'].get('dedicatedHost', None)]
-    ded_cpus = [int(x['template']['startCpus']) for x in result['processors']
+    ded_cpus = [int(x['template']['startCpus']) for x in create_options['processors']
                 if x['template'].get('dedicatedAccountHostOnlyFlag', False)]
-    ded_host_cpus = [int(x['template']['startCpus']) for x in result['processors']
+    ded_host_cpus = [int(x['template']['startCpus']) for x in create_options['processors']
                      if x['template'].get('dedicatedHost', None)]
 
     standard_cpus = sorted(standard_cpus)
-    table.add_row(['cpus (standard)', formatting.listing(standard_cpus, separator=',')])
+    cpu_table.add_row(['standard', formatting.listing(standard_cpus, separator=',')])
     ded_cpus = sorted(ded_cpus)
-    table.add_row(['cpus (dedicated)', formatting.listing(ded_cpus, separator=',')])
+    cpu_table.add_row(['dedicated', formatting.listing(ded_cpus, separator=',')])
     ded_host_cpus = sorted(ded_host_cpus)
-    table.add_row(['cpus (dedicated host)', formatting.listing(ded_host_cpus, separator=',')])
+    cpu_table.add_row(['dedicated host', formatting.listing(ded_host_cpus, separator=',')])
+    return cpu_table
+
 
-    # Memory
-    memory = [int(m['template']['maxMemory']) for m in result['memory']
+def _get_memory_table(create_options):
+    memory_table = formatting.Table(['memory', 'value'], title='Memories')
+    memory_table.sortby = 'memory'
+    memory_table.align = 'l'
+    memory = [int(m['template']['maxMemory']) for m in create_options['memory']
               if not m['itemPrice'].get('dedicatedHostInstanceFlag', False)]
-    ded_host_memory = [int(m['template']['maxMemory']) for m in result['memory']
+    ded_host_memory = [int(m['template']['maxMemory']) for m in create_options['memory']
                        if m['itemPrice'].get('dedicatedHostInstanceFlag', False)]
 
     memory = sorted(memory)
-    table.add_row(['memory',
-                   formatting.listing(memory, separator=',')])
+    memory_table.add_row(['standard',
+                          formatting.listing(memory, separator=',')])
 
     ded_host_memory = sorted(ded_host_memory)
-    table.add_row(['memory (dedicated host)',
-                   formatting.listing(ded_host_memory, separator=',')])
+    memory_table.add_row(['dedicated host',
+                          formatting.listing(ded_host_memory, separator=',')])
+    return memory_table
 
-    # Operating Systems
+
+def _get_os_table(create_options):
```

Comment:
to make this a bit simpler, lets do the following


```
def _get_os_table(create_options):
    os_table = formatting.Table(['KeyName', 'Description'], title='Operating Systems')
    os_table.sortby = 'KeyName'
    os_table.align = 'l'
    op_sys = []
    for operating_system in create_options['operatingSystems']:
        os_option = {
            'referenceCode': operating_system['template']['operatingSystemReferenceCode'],
            'description': operating_system['itemPrice']['item']['description']
        }
        op_sys.append(os_option)


    for operating_system in op_sys:
        os_table.add_row([
            operating_system['referenceCode'],
            operating_system['description']
        ])
    return os_table
```

Reply:
Done